source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -192,6 +193,14 @@ namespace PlayFab
                     {
                         // Add HTTP headers
                         SetPredefinedHeaders(reqContainer, hRequest);
+                        if(!headerNoErrors)
+                        {
+                            SetErrorInfo(reqContainer, "Error in attempting to add Default Headers with HRESULT: " + std::to_string(GetLastError()));
+                            CloseOutConnections(std::move(requestContainer), hRequest, hConnect, hSession);
```

Comment:
nit: it makes it difficult to read that some of the codepaths for `ExecuteRequest` return early on failure whereas other failure codepaths just bubble down to the final CloseOutConnections call by virtue of nested if-else blocks.

it doesn't help that ExecuteRequest is a large function with lots of nesting.

recommend restructuring ExecuteRequest to either use early returns for all failure paths or to not use them at all.

Frankly, I think it would be a general readability improvement to use early returns for all failure cases, but I'll leave that to you and the SDK teams preference (maybe y'all like the nesting. some members of my team sure do). There are a number of ways to enable early returns for all codepaths. You could just explicitly call CloseOutConnections at each failure point or you could turn ExecuteRequest into a wrapper that calls two helper functions, SendRequest and CompleteRequest

```
void ExecuteRequest(...)
{
     HINTERNET hSession = NULL, hConnect = NULL, hRequest = NULL;
     TrySendRequest(*callRequestContainer, &hSession, &hConnect, &hRequest);
     CompleteRequest(std::move(callRequestContainer), hSession, hConnect, hRequest));
}

void
TrySendRequest(CallRequestContainer& reqContainer, HINTERNET* phSession, HINTERNET* phConnect, HINTERNET* phRequest)
{
    bool res = SomeOperation(..);
    if (!res)
    {
        SetErrorInfo(blah blah balh);
        return;
    }
    /// ....
}

void
CompleteRequest(std::unique_ptr<CallRequestContainer> reqContainer, HINTERNET hSession, HINTERNET hConnect, HINTERNET hRequest)
{
   // .. this is just your CloseOutConnection helpers which I've renamed for this context
}
```